CONTRIBUTING.md

```diff
@@ -0,0 +1,36 @@
+For the general contribution guidelines see
+[fractalyze/CONTRIBUTING](https://github.com/fractalyze/.github/blob/main/CONTRIBUTING.md).
+
+## Updating LLVM upstream patches
+
+When touching files under `third_party/llvm-project`, follow this process to
+refresh the patches we carry in this repository.
+
+1. Run `tools/setup_llvm_clone.sh /absolute/path/to/local/llvm-project` to clone
+   the canonical revision (it creates that directory if it does not already
+   exist) and apply the current `zkir` patches automatically. The script accepts
+   an existing clone as well, in which case it just checks out the correct
+   commit and reapplies the patches.
+
+1. Point Bazel to the local clone while you iterate:
+
+   - Uncomment the `http_archive(name = "llvm-raw", …)` stanza in
+     `WORKSPACE.bazel`.
+
+   - Add (or uncomment) the `new_local_repository` entry:
+
+     ```
+     new_local_repository(
+         name = "llvm-raw",
+         build_file_content = "# empty",
+         path = "/absolute/path/to/your/llvm-project",
+     )
+     ```
+
+1. Implement and test your LLVM changes.
+
+1. Produce an updated patch and copy it back under `third_party/llvm-project/`:
+
+   ```sh
+   git show HEAD --no-prefix > /path/to/zkir/repo/third_party/llvm-project/<descriptive_name>.patch
+   ```
```

Comment:
I think the important thing you need to mention is `--no-prefix`! Since other than `git show HEAD`, there are a lot of way to produce a diff `git diff`, `git diff --cached`!